docs/src/main/asciidoc/security-webauthn.adoc

```diff
@@ -668,13 +566,44 @@ image::webauthn-4.png[role="thumb"]
 
 The admin API is only accessible if you register with the `admin` user name.
 
+== Explanation of the flow for WebAuthn authentication.
+
+There are two modes of operation for how you want to handle user and credential creation:
+
+By default, the solution we have seen handles user and credential creation and logins automatically
+via the <<WebAuthn endpoints>> provided out of the box by this extension.
+
+In this scenario, all you have to to is provide a link:{webauthn-api}/io/quarkus/security/webauthn/WebAuthnUserProvider.html[`WebAuthnUserProvider`] to
+tell Quarkus how to find, store and update your credentials (see 
+<<Exposing your entities to Quarkus WebAuthn>>).
+
+Thus, triggering a registration on the client requires obtaining a challenge from the server (see
+<<Obtain a registration challenge>>), which will then be returned to the client which
+uses that to create new credentials and send them to the server (see
+<<Trigger a registration>>) which will store the credentials and log you in:
+
+image::webauthn-register.svg[role="thumb"]
+
+Later, triggering a login on the client requires obtaining a challenge from the server (see
+<<Obtain a login challenge>>), which will then be returned to the client which
+uses that to get the required credentials and send proof of them to the server (see
+<<Trigger a login>>) which will update the credentials and log you in:
+
+image::webauthn-login.svg[role="thumb"]
+
+If you want to customise the registration and login flow, you can also handle user and credential
+creation and logins yourself via your own logic (see <<Handling login and registration endpoints yourself>>).
+
 == WebAuthn endpoints
 
 The Quarkus WebAuthn extension comes out of the box with these REST endpoints pre-defined:
 
 === Obtain a registration challenge
 
-`POST /q/webauthn/register`: Set up and obtain a registration challenge
+`POST /q/webauthn/register-challenge`: Set up and obtain a registration challenge
+
+This causes two cookies to be set: one for the challenge, and one for the user name, they
```

Comment:
@FroMage Is the user name and challenge submissions have different lifecycles ? Ex, you submit the challenge, and then from some other page the user name ? Trying to understand  the rationale for using 2 cookies...

Reply:
This is due to just not having Quarkus sessions stored in cookies. I could unify them in a single cookie, but they were always two cookies.

Reply:
@FroMage Sorry, I don't understand it. Can you please expand a bit more on how these 2 cookies are used ?

Reply:
@FroMage In OIDC, we also have 2 different cookies, one for the actual flow to complete, the state cookie, and the session cookie, and I suppose something similar is happening here, while the registration is ongoing, you have one cookie to coordinate it and then once the reg is complete, the actual session cookie (you refer to it `one for the user name`)...
This is all fine then, but IMHO you should either avoid talking about cookies or give more specific information, which cookie is created when, do they have the same lifecycle, names (for the users who extend the default solution) to know how to handle them... 
Thanks

Reply:
The cookies are used to store a challenge and username, because the webauthn protocol requires to obtain a challenge from the server before calling the login/registration endpoint.